### ISSUE TITLE
[Rockon] when adding storage pre-addition summary screen displays incorrectly

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2933]
As a continuation of #2904 that was fixed by #2914:

Thanks to forum user stevek who observed a similar behavior when adding storage to an installed Rockon:

https://forum.rockstor.com/t/syncthing-add-storage-function/10321/

I assume, it's probably the same root cause, just in a different place of the coding.

**Comment by FroggyFlox** [comment]
Following the description of the issue in the forum thread linked above, we indeed see the following as the summary of the changes to be instantiated after the "Add Storage" procedure:
![image](https://github.com/user-attachments/assets/f9ec5274-54f4-4a2a-911d-9e5583dc9e04)


The Backbone template is https://github.com/rockstor/rockstor-core/blob/812f1cffe39b0c7b6005215355b7fa413feae4ea/src/rockstor/storageadmin/static/storageadmin/js/templates/rockons/settings_summary_table.jst#L17

The display of newly-added volumes is thus defined by the handlebars helper: `display_newVolumes`, itself defined in:
https://github.com/rockstor/rockstor-core/blob/812f1cffe39b0c7b6005215355b7fa413feae4ea/src/rockstor/storageadmin/static/storageadmin/js/views/rockons.js#L1617-L1630

We could thus swap the 2 rows in questions to fix the bug, but this would most likely just cover the real source of the "bug" here. We indeed, use the construct for `newLabels` and `newCnets` and these are fine. We should thus probably fix the mapping of share:internal_representation.

**Comment by FroggyFlox** [comment]
> We should thus probably fix the mapping of share:internal_representation.

This is defined at:
https://github.com/rockstor/rockstor-core/blob/812f1cffe39b0c7b6005215355b7fa413feae4ea/src/rockstor/storageadmin/static/storageadmin/js/views/rockons.js#L1202-L1204

Swapping the key:value here as follows...
```js
        this.share_map = this.model.get('shares');
        // this.share_map[this.$('#volume').val()] = this.$('#share').val();
        this.share_map[this.$('#share').val()] = this.$('#volume').val();
        this.model.set('shares', this.share_map);
```

... then rebuilding the static files...
```
cd /opt/rockstor
poetry run django-admin collectstatic
```

... and we now have:
![image](https://github.com/user-attachments/assets/7606a09b-dbae-4746-9957-322dba2f2274)

**Comment by FroggyFlox** [comment]
While the table is now displayed, properly, the **update process itself fails**, however:
![image](https://github.com/user-attachments/assets/3cc541ed-a772-4a48-9dea-64b4de9790c5)

This is due to the fact that our backend is currently expecting the dict to be
```
key: volume_name
value: share_name
```

While it would be more appropriate to be consistent across our various update objects here, the amount of refactoring required in our backend does not fit the current very late stage in our testing phase. I thus favor the first option laid out early in this thread: adjusting the `display_newVolumes` handlebars helper. @phillxnet, would you agree here?

**Comment by phillxnet** [comment]
@FroggyFlox Re:

> While it would be more appropriate to be consistent across our various update objects here, the amount of refactoring required in our backend does not fit the current very late stage in our testing phase. I thus favor the first option laid out early in this thread: adjusting the `display_newVolumes` handlebars helper. @phillxnet, would you agree here?

Agreed. We can address the referenced inconsistencies when we get to switching out some of our interface technologies. As you say, we should try to avoid deeper changes at the end of a testing phase.

**Comment by FroggyFlox** [comment]
Thanks for the feedback!
I'll move the associated PR from "draft" to "ready" now that we agree on the approach, then.

**Comment by FroggyFlox** [comment]
Closing as fixed by #2942.
